x/evm/vm/interface.go

```diff
@@ -0,0 +1,52 @@
+package vm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+// PrecompiledContracts defines a map of address -> precompiled contract
+type PrecompiledContracts map[common.Address]vm.PrecompiledContract
+
+type StatefulPrecompiledContract interface {
+	vm.PrecompiledContract
+	RunStateful(evm EVM, addr common.Address, input []byte, value *big.Int) (ret []byte, err error)
+}
+
+// EVM defines the interface for the Ethereum Virtual Machine used by the EVM module.
+type EVM interface {
+	Config() vm.Config
+	Context() vm.BlockContext
+	TxContext() vm.TxContext
+
+	Reset(txCtx vm.TxContext, statedb vm.StateDB)
+	Cancel()
+	Cancelled() bool //nolint
+	Interpreter() *vm.EVMInterpreter
+	Call(caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error)
+	CallCode(caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error)
+	DelegateCall(caller vm.ContractRef, addr common.Address, input []byte, gas uint64) (ret []byte, leftOverGas uint64, err error)
+	StaticCall(caller vm.ContractRef, addr common.Address, input []byte, gas uint64) (ret []byte, leftOverGas uint64, err error)
+	Create(caller vm.ContractRef, code []byte, gas uint64, value *big.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error)
+	Create2(caller vm.ContractRef, code []byte, gas uint64, endowment *big.Int, salt *uint256.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error)
```

Comment:
I am not sure how `Create2` is supposed to be different from Create but either the name should have that info somehow or this function should have a description.

Reply:
@emailtovamos Thanks for your thoughts on this. I must disagree with you, since [CREATE2](https://www.evm.codes/#f5) is well-know EVM op_code using for deploy contracts at a consistent, deterministic address. So, It is not necessary to extra describe it. Also exact name is necessary for EVM interface implementation.